frontend/apps/reader/modules/readerpaging.lua

```diff
@@ -1062,8 +1062,21 @@ function ReaderPaging:_gotoPage(number, orig_mode)
     return true
 end
 
-function ReaderPaging:onGotoPage(number)
+function ReaderPaging:onGotoPage(number, pos)
     self:_gotoPage(number)
+    self:setPagePosition(number, 0)
+    if pos then
+        local rect_p = Geom:new{ x = pos.x or 0, y = pos.y or 0 }
+        local rect_s = Geom:new(rect_p):copy()
+        rect_s:transformByScale(self.view.state.zoom)
+        if self.view.page_scroll then
+            self:onScrollPanRel(rect_s.y - self.view.page_area.y)
+        else
+            self.view:PanningUpdate(rect_s.x - self.view.visible_area.x, rect_s.y - self.view.visible_area.y)
+        end
+    else
+        self.ui:handleEvent(Event:new("PageUpdate", self.current_page))
```

Comment:
Btw, how does this interact with go to previous location? Or perhaps you'd also want to store position data in there?

Reply:
Jumping back to the previous location worked fine when using the "Go to page" feature and when following links. The latter is my main use case, as I usually want to see a reference and then jump back.

Reply:
Yes, that's what it's (mainly) for. But I've never noticed an issue (in fact if you asked me I might swear it took me to the location on the page, lol), so you're saying the issue you're fixing doesn't exist in reverse?

Reply:
The original behaviour:
1. In page mode: upon page change, go to top. If the page does not change, do nothing.
2. In continuous mode: this seems to restore the state of the destination page, which defaults to the top if it wasn't set/modified before.
Jumping back simply restores the page and position, I assume, which would explain why it "doesn't exist in reverse."

Reply:
The restore location functionality actually first restores positions and then sets the page: https://github.com/koreader/koreader/blob/357bc65217ed7bdfaae6a433449d0f21baa6abc5/frontend/apps/reader/modules/readerpaging.lua#L597-L605
It also tracks whether a `PageUpdate` event is required or not. Should I adapt the same approach?

Reply:
It would be my natural inclination to use the same pattern, unless there's a reason not to. If you don't mind. ^_^

Reply:
Sure, already done.